[DOC] Tokenizers - Classic (#8357)
opensearch-project/documentation-website: _analyzers/tokenizers/classic.md

---
layout: default
title: Classic
parent: Tokenizers
nav_order: 35

---

# Classic tokenizer

> The `classic` tokenizer parses text, applying the English language grammatical rules to break the text into tokens. It includes specific logic to handle patterns such as the following:

Reviewer: ```suggestion
The `classic` tokenizer parses text, applying English language grammatical rules to break the text into tokens. It includes specific logic to handle patterns such as the following:
```

- acronyms 
- email addresses
- domain names
- certain types of punctuation

> This tokenizer works best with the English language. It may not produce optimal results for other languages, especially those with different grammatical structures.

Reviewer 2: Reworded line 11.

{: .note}

The `classic` tokenizer parses text as follows:

> - **Punctuation**: Splits text at most punctuation marks and removes punctuation characters. Dots that aren't followed by spaces are treated as part of the token.

Reviewer 2: ```suggestion
- **Punctuation**: Splits text on most punctuation marks and removes punctuation characters. Dots that aren't followed by spaces are treated as part of the token.
```

> - **Punctuation**: Splits text at most punctuation marks and removes punctuation characters. Dots that aren't followed by spaces are treated as part of the token.

Reviewer 2: Changed to "split on"

> - **Hyphens**: Splits words at hyphens, except when a number is present. When a number is present in a token, the token is not split and is treated like a product number. 

Reviewer: Like a "product number" specifically?

Reviewer 2: Yes

    },
    {
      "token": "number",
      "start_offset": 99,
      "end_offset": 105,
      "type": "<ALPHANUM>",
      "position": 14
    },
    {
      "token": "1-800-555-1234",
      "start_offset": 106,
      "end_offset": 120,
      "type": "<NUM>",
      "position": 15
    },
    {
      "token": "P.S.",
      "start_offset": 122,
      "end_offset": 126,
      "type": "<ACRONYM>",
      "position": 16
    },
    {
      "token": "你",
      "start_offset": 127,
      "end_offset": 128,
      "type": "<CJ>",
      "position": 17
    },
    {
      "token": "好",
      "start_offset": 128,
      "end_offset": 129,
      "type": "<CJ>",
      "position": 18
    }
  ]
}
```

## Token types

The `classic` tokenizer produces the following token types.

| Token type    | Description  | 
| :--- | :--- | 
| `<ALPHANUM>`  | Alphanumeric tokens consisting of letters, numbers, or a combination of both.                     | 
| `<APOSTROPHE>`| Tokens containing an apostrophe, commonly used in possessives or contractions (for example, `John's`).   |
| `<ACRONYM>`   | Acronyms or abbreviations, often identified by a trailing period (for example, `P.S.` or `U.S.A.`).     |

> | `<COMPANY>`   | Tokens representing company names (for example, `X&Y`). If these tokens aren’t produced automatically, you may need custom configurations or filters.  | 

Reviewer: ```suggestion
| `<COMPANY>`   | Tokens representing company names (for example, `X&Y`). If these tokens aren't produced automatically, you may need custom configurations or filters.  | 
```

> | `<EMAIL>`     | Tokens that match email addresses, containing an `@` symbol and a domain (for example,`support@widgets.co` or `info@example.com`). |

Reviewer: ```suggestion
| `<EMAIL>`     | Tokens matching email addresses, containing an `@` symbol and a domain (for example,`support@widgets.co` or `info@example.com`). |
```

| `<HOST>`      | Tokens matching website or host names, often containing `www.` or a domain suffix like `.com` (for example, `www.example.com` or `example.org`).  |

> | `<NUM>`       | Purely numeric tokens or numeric-like sequences (for example, `1-800`, `12345`, or `3.14`).     |

Reviewer: ```suggestion
| `<NUM>`       | Tokens containing only numeric tokens or numeric-like sequences (for example, `1-800`, `12345`, or `3.14`).     |
```

Reviewer 2: Edited suggestion.

| `<CJ>`        | Tokens representing Chinese or Japanese characters.   |

> | `<ACRONYM_DEP>` | Deprecated acronym handling (for example, acronyms with different parsing rules in older versions). Rarely used; exists primarily for backward compatibility with legacy tokenizer rules. | 

Reviewer: ```suggestion
| `<ACRONYM_DEP>` | Deprecated acronym handling (for example, acronyms with different parsing rules in older versions). Rarely used---exists primarily for backward compatibility with legacy tokenizer rules. | 
```
